### ISSUE TITLE
Is there a way to test your endpoint (or download the XML file)?

### ISSUE THREAD

**Issue by gregblass** [qbwc/qbwc#105]
I've got what I think is a valid implementation going. Is there any way I can test it now without deploying it and seeing if it successfully goes to the client's actual quickbooks account?

Also/or, is there a way to at least download the XML file that will be sent once a job is queued in the database? At least that way I can load it into the SDK checker program that they have?

Or maybe another way would be to have my client create another company file and try to interface with it there?

Or - I'm just creating a deposit in my case - I guess I could just have them delete the test deposit after I see if it worked?

Does anyone have any insight on how they tackled this type of thing? Much appreciated!

**Comment by JasonBarnabe** [comment]
Personally, I made a copy of the Quickbooks file and hooked up to that to do my testing.

**Comment by gregblass** [comment]
So you installed quickbooks on your local machine? I guess I could install it on virtual box. I really don't want to buy quickbooks just to test this. Or pirate it.

**Comment by JasonBarnabe** [comment]
No, I remoted in to one of the client's computers to test, but this is getting off topic. To be able to test that it works in Quickbooks, you will need a machine running Quickbooks. There is probably a way to get the XML but I wouldn't rely on the SDK checker as there are a number of potential issues that it would not be able to detect, like the existence of the various accounts and such being referenced.

**Comment by gregblass** [comment]
I just called the quickbooks sales number, and they give out "NFR" (not for resale) copies of quickbooks for developers if you ask for it. So problem solved!

I'd still like to know how to download a copy of the XML thats being generated though.

**Comment by JasonBarnabe** [comment]
Some combination of

https://github.com/qbwc/qbwc/blob/master/lib/qbwc/session.rb#L43
https://github.com/qbwc/qbwc/blob/master/lib/qbwc/request.rb#L20

should do it, but I've never tried.

**Comment by shaofenchen** [comment]
Maybe SoapUI? Get the WSDL for QB web connector and import it into SoapUI project to create request templates to act as the QBWC. Then start poking around from there.

**Comment by Rina-Y** [comment]
I set Quickbooks Web Connector registry key to VERBOSE from DEBUG, it will put the entire request and stuff into its log (if you do this, increase the size of the log too).  HKEY_CURRENT_USER\Software\Intuit\QBWebConnector
Level to VERBOSE and LogFileSize to 0x030d4000 (51200000, i added 2 zeros to the decimal version)

You can also check what it would output by using the QBWC.parser.to_xml(your-hash) method. I did this a bunch in rails console (irb)
